### PR TITLE
Certik severity informational

### PR DESCRIPTION
**Pull request by asalzmann**
## What is the purpose of the change

Fixes a variety of informational issues flagged in the CertiK audit. Commits are linked to the audit feedback.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? audit

### CODE REVIEW

File: x/stakeibc/module.go
Comment (riley-stride) [inline x/stakeibc/module.go:177]: Can you explain this change? Seems unrelated to SBU-03

Comment (riley-stride) [inline x/stakeibc/module.go:177]: Can you explain this change? Seems unrelated to SBU-03

**Review by riley-stride — COMMENTED** [review]
LGTM overall, only thing before merging: seems like some outstanding nit fixed on SBU-03, waiting for them to come back and clarify which files are in scope. Shall we hold off on merging until then?

File: x/stakeibc/module.go
Comment (asalzmann) [reply x/stakeibc/module.go:177]: I also don't see it in the audit report - they might have changed it? Pretty sure this was flagged 🤔 the code path is unused, but I can revert this change if it makes more sense

Comment (riley-stride) [reply x/stakeibc/module.go:177]: The current audit report is the newest version, which is updated. The old one is in [audit channel](https://slack.com/archives/#C03MRQZV31S) as a pdf if you wanna diff [](edited-with-Axolo)

**Review by riley-stride — APPROVED** [review]
Approved, subject to reverting https://github.com/Stride-Labs/stride/pull/145/files/6dd11dc4ec07a0ab6add2796906adbb8d4a8c12a..bd2f7438df74fca11d4e534748717c90358b7860#diff-0954b90851fb39debd1c8123f62141c9111109aee55ef6b157773a35361e74fbL177 which does not make sense to me, per comments

**Comment by asalzmann** [comment]
The `BeginBlocker` function doesn't do anything, so I think it's safe to remove. I'll see if I can find the certik comment

File: x/stakeibc/module.go
Comment (asalzmann) [reply x/stakeibc/module.go:177]: Can't find it in the older audit report, I'll just revert this

**Pull request merged**
